src/core/net/nd6.cpp

```diff
@@ -170,6 +170,16 @@ uint8_t RouteInfoOption::OptionLengthForPrefix(uint8_t aPrefixLength)
     return length;
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+// RaFlagsExtOption
+
+void RaFlagsExtOption::Init(void)
+{
+    Clear();
+    SetType(kTypeRaFlagsExtension);
+    SetSize(sizeof(*this));
```

Comment:
The existing PIO uses the following style: `SetSize(sizeof(PrefixInfoOption));`
Is there a reason to do it differently here? Probably no difference in the compile result.

Reply:
Both ways are fine. 

Some folks like/prefer `sizeof(*this)` as it is less likely to get it wrong (using a wrong/different type name in `sizeof()` call). I myself use both.